File: server/nodejs/src/payment/paymentDetailsService.js
const config = require('../configuration/config')
const paymentSdk = require('../configuration/paymentSdk')

/**
 * Invokes a getPaymentDetails request towards the paymentSdk API for given payment id
 */
exports.getPaymentDetails = async (paymentId) => {
    const paymentDetails = await paymentSdk.payments.getPaymentDetails(config.merchantId, paymentId, {});
    return paymentDetails;
}

/**
 * Invokes a getPaymentDetails request towards the paymentSdk API for given hosted checkout id
 */
exports.getPaymentDetailsForHostedCheckout = async (hostedCheckoutId) => {
    const paymentId = `${hostedCheckoutId}_0`;
    return await this.getPaymentDetails(paymentId);
}

File: server/python/payment/payment_details_service.py
from configuration.merchant_client_config import merchant_client

def get_payment_details(payment_id):
    return merchant_client().payments().get_payment_details(payment_id)

def get_payment_details_for_hosted_checkout(hosted_checkout_id):
    payment_id = f'{hosted_checkout_id}_0'
    return merchant_client().payments().get_payment_details(payment_id)